src/Command/Environment/EnvironmentXdebugCommand.php

```diff
@@ -0,0 +1,106 @@
+<?php
+namespace Platformsh\Cli\Command\Environment;
+
+use Platformsh\Cli\Command\CommandBase;
+use Platformsh\Cli\Service\Ssh;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Process\Process;
+
+class EnvironmentXdebugCommand extends CommandBase
+{
+    const SOCKET_PATH = '/run/xdebug-tunnel.sock';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('environment:xdebug')
+            ->setAliases(['xdebug'])
+            ->addOption('port', null, InputArgument::OPTIONAL, 'The local port for Xdebug to connect to.', 9000)
+            ->setDescription('Reverse tunnel Xdebug to the current host');
+        $this->addProjectOption()
+             ->addEnvironmentOption()
+             ->addRemoteContainerOptions();
+        Ssh::configureInput($this->getDefinition());
+        $this->addExample('Connect the environment to Xdebug listening locally on port 9000.');
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->validateInput($input);
+        $this->getSelectedEnvironment();
+
+        $container = $this->selectRemoteContainer($input);
+        $sshUrl = $container->getSshUrl();
+
+        $config = $container->getConfig()->getNormalized();
+        $key = isset($config['runtime']['xdebug']['key']) ? $config['runtime']['xdebug']['key'] : '';
+
+        if (!$key) {
+            $output->getErrorOutput()->writeln(
+                "<error>A debugging key has not been found</error>\n" .
+                "\n" .
+                "To use Xdebug your project must have a <info>debugging key</info> set.\n" .
+                "Such key is set in the <info>.platform.app.yaml</info> file as in this example:\n" .
```

Comment:
Does this need per-vendor thing?

Reply:
Yes, `$this->config()->get('service.app_config_file')`